### ISSUE TITLE
Wrong edge labels in plot_complexes 

### ISSUE THREAD

**Issue by andreasnoack** [SciML/Catalyst.jl#1135]
**Describe the bug 🐞**

The printed edge labels are wrong. The labels are printed with the wrong edges.

**Expected behavior**

I expect the edges to have the correct labels

**Minimal Reproducible Example 👇**

Here is a reproducer based on a small pharmacokinetic example

```julia
using Catalyst, GraphMakie, WGLMakie

rs = @reaction_network begin
    ka, Depot --> Central
    (k12, k21), Central <--> Peripheral
    ke, Central --> 0
end

Catalyst.plot_network(rs)

f, ax, p = Catalyst.plot_complexes(rs)

hidedecorations!(ax); hidespines!(ax); ax.aspect = DataAspect()

f
```

**Error & Stacktrace ⚠️**

<img width="287" alt="Skærmbillede 2024-11-25 kl  08 59 18" src="https://github.com/user-attachments/assets/57e6e700-63da-4fda-ac32-2dcbd41b38e8">

**Environment (please complete the following information):**

  - Output of `using Pkg; Pkg.status()`

```julia

```

  - Output of `using Pkg; Pkg.status(; mode = PKGMODE_MANIFEST)`

```julia
(TestGraphMakie) pkg> st
Status `~/Desktop/TestGraphMakie/Project.toml`
  [479239e8] Catalyst v14.4.1 `~/.julia/dev/Catalyst`
  [1ecd5474] GraphMakie v0.5.12
  [86223c79] Graphs v1.12.0
  [276b4fcb] WGLMakie v0.10.16
```

  - Output of `versioninfo()`

```julia
julia> versioninfo()
Julia Version 1.10.5
Commit 6f3fdf7b362 (2024-08-27 14:19 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 8 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 2 default, 0 interactive, 1 GC (on 6 virtual cores)
Environment:
  JULIA_NUM_THREADS = 2
  JULIA_EDITOR = code
```

**Additional context**

I suspect this has to do with the iteration order of `edges(rs)`, since
```julia
julia> collect(edges(_img))
4-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 1 => 2
 Edge 2 => 3
 Edge 2 => 4
 Edge 3 => 2

julia> reactions(rs)
4-element Vector{Reaction}:
 ka, Depot --> Central
 k12, Central --> Peripheral
 k21, Peripheral --> Central
 ke, Central --> ∅
```

**Comment by isaacsas** [comment]
@vyudu

**Comment by isaacsas** [comment]
Thanks for reporting!

**Comment by devmotion** [comment]
It seems the issue was fixed by #1138?

**Comment by isaacsas** [comment]
Closed by https://github.com/SciML/Catalyst.jl/pull/1138
